### PR TITLE
Add DE10-Nano-based MiSTer

### PR DESCRIPTION
**Pull request by Ravenslofty**
It'd be nice to have some way for this to inherit from `DE10NanoPlatform`, but oh well.

### CODE REVIEW

File: nmigen_boards/mister.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            gpio_0:26 gpio_0:23 gpio_0:24 gpio_0:21 gpio_0:36 gpio_0:22 gpio_0:19
+            """, dq="""
+            gpio_0:1  gpio_0:2  gpio_0:3  gpio_0:4  gpio_0:5  gpio_0:6  gpio_0:7  gpio_0:8
+            gpio_0:18 gpio_0:17 gpio_0:16 gpio_0:15 gpio_0:14 gpio_0:13 gpio_0:9  gpio_0:10
+            """, dqm="", attrs=Attrs(io_standard="3.3-V LVCMOS")),
+
+        # MiSTer I/O Board (optional, but highly recommended)
+        # https://github.com/MiSTer-devel/Hardware_MiSTer/blob/master/releases/iobrd_6.0.pdf
+        Resource("power_led", 0, PinsN("1", dir="o", conn=("gpio", 1)), Attrs(io_standard="3.3-V LVTTL")),
+        Resource("disk_led", PinsN("3", dir="o", conn=("gpio", 1)), Attrs(io_standard="3.3-V LVTTL")),
```
Comment (whitequark) [inline nmigen_boards/mister.py]: I think these don't actually work, since you did not specify the number.

**Comment by whitequark** [comment]
I've added the `conn` argument to all factory functions.

File: nmigen_boards/mister.py [hunk 2]
```diff
@@ -65,44 +65,40 @@ class MisterPlatform(IntelPlatform):
...
+            conn=("gpio_1", 1), attrs=Attrs(io_standard="3.3-V LVTTL")),
 
         # The schematic is difficult to understand here...
         Resource("vga", 0,
-            Subsignal("r", Pins("28 32 34 36 38 40", dir="o", conn=("gpio", 1))),
-            Subsignal("g", Pins("27 31 33 35 37 39", dir="o", conn=("gpio", 1))),
-            Subsignal("b", Pins("21 23 25 26 24 24", dir="o", conn=("gpio", 1))),
-            Subsignal("hs", Pins("20", dir="o", conn=("gpio", 1))),
-            Subsignal("vs", Pins("19", dir="o", conn=("gpio", 1))),
+            Subsignal("r", Pins("28 32 34 36 38 40", dir="o", conn=("gpio_1", 1))),
```
Comment (whitequark) [inline nmigen_boards/mister.py]: Shouldn't this be `conn=("gpio", 1)`?

Comment (Ravenslofty) [reply nmigen_boards/mister.py]: Ah, I got confused with your other comment. I'll probably fix this tomorrow.

**Comment by Ravenslofty** [comment]
I'm an idiot and forgot to push my fix commit.

**Comment by whitequark** [comment]
Thanks!

**Pull request merged**